packages/wallet/backend/src/account/service.ts

```diff
@@ -70,51 +61,25 @@ export class AccountService implements IAccountService {
 
     // issue virtual account to wallet
     const user = await User.query().findById(args.userId)
-    if (!user) {
+    if (!user || !user.gateHubUserId) {
       throw new NotFound()
     }
 
-    const result = await this.rapydClient.issueVirtualAccount({
-      country: user.country ?? '',
-      currency: asset.code,
-      ewallet: user.rapydWalletId ?? ''
-    })
-
-    if (result.status?.status !== 'SUCCESS') {
-      throw new Error(
-        `Unable to issue virtal account to ewallet: ${result.status?.message}`
-      )
-    }
-    // save virtual bank account number to database
-    const virtualAccount = result.data
+    const result = await this.gateHubClient.createWallet(
+      user.gateHubUserId,
+      args.name
+    )
 
     const account = await Account.query().insert({
       name: args.name,
       userId: args.userId,
       assetCode: asset.code,
       assetId: args.assetId,
       assetScale: asset.scale,
-      virtualAccountId: virtualAccount.id
-    })
-    await this.rapydClient.simulateBankTransferToWallet({
-      amount: 0,
-      currency: account.assetCode,
-      issued_bank_account: account.virtualAccountId
+      gateHubWalletId: result.address
     })
 
-    if (!user || !user.rapydWalletId) {
-      throw new NotFound()
-    }
-
-    const accountsBalance = await this.rapydClient.getAccountsBalance(
-      user.rapydWalletId
-    )
-
-    account.balance = transformBalance(
-      accountsBalance.data?.find((acc) => acc.currency === account.assetCode)
-        ?.balance ?? 0,
-      asset.scale
-    )
+    account.balance = transformBalance(0, account.assetScale)
```

Comment:
Looks like this will also be handled in a different PR as well? Or should the balance be 0 at the account creation?

Reply:
at account creation the balance is always 0